envoy-control-core/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/snapshot/SnapshotProperties.kt

```diff
@@ -67,6 +67,7 @@ class IncomingPermissionsProperties {
     var enabled = false
     var clientIdentityHeaders = listOf("x-service-name")
     var trustedClientIdentityHeader = "x-client-name-trusted"
+    var requestIdentificationHeaders = listOf("x-custom-request-id")
```

Comment:
This is one of the ways this can be done. Envoy as a default would add x-request-id header, so when I enabled this functionality all other tests that use incoming permissions logs would fail. The way I avoided that is that I changed the default header here. Other ways could include:

Always setting x-request-id to a custom value when doing requests in tests using incoming permissions logs and assert on that custom value. I don't like this approach because it could easily be forgot when writing new test.
Changing the way we assert on incoming permissions logs, not taking into account requestId and doing it only via a flag. I do not like this approach because it complicates the assertion logic.